Source/IntegrationTests/TestFiles/LitTests/LitTest/ast/overrides.dfy

```diff
@@ -0,0 +1,43 @@
+// RUN: %testDafnyForEachCompiler %s
+ 
+trait T {
+  method foo() returns (r: int)  {
+    return 3;
+  }
+}
+
+trait T2 extends T {
+  method foo() returns (r: int)  {
+    return 5;
+  }
+}
+
+class C extends T {
+  method foo() returns (r: int) {
+    return 4;
+  }
+}
+
+class D extends T {}
+class E extends T2 {}
+class F extends T2 {
+  method foo() returns (r: int) {
+    return 6;
+  }
+}
+
+method Main() {
+  var c := new C;
+  var rc := c.foo();
+  var d := new D;
+  var rd := d.foo();
+  var e := new E;
+  var re := e.foo();
+  var f := new F;
```

Comment:
I would add `: F` as well, just in case type inference gets the crazy idea to make the type of `f` `T2` as well.